pkg/libartifact/store/store.go

```diff
@@ -0,0 +1,375 @@
+//go:build !remote
+
+package store
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/common/libimage"
+	"github.com/containers/image/v5/manifest"
+	"github.com/containers/image/v5/oci/layout"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/podman/v5/pkg/libartifact"
+	libartTypes "github.com/containers/podman/v5/pkg/libartifact/types"
+	"github.com/containers/storage"
+	"github.com/go-openapi/runtime"
+	"github.com/opencontainers/go-digest"
+	"github.com/opencontainers/image-spec/specs-go"
+	specV1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	ErrEmptyArtifactName = errors.New("artifact name cannot be empty")
+)
+
+type ArtifactStore struct {
+	SystemContext *types.SystemContext
+	storePath     string
+}
+
+// NewArtifactStore is a constructor for artifact stores.  Most artifact dealings depend on this. Store path is
+// the filesystem location.
+func NewArtifactStore(storePath string, sc *types.SystemContext) (*ArtifactStore, error) {
+	// storePath here is an override
+	if storePath == "" {
+		storeOptions, err := storage.DefaultStoreOptions()
+		if err != nil {
+			return nil, err
+		}
+		if storeOptions.GraphRoot == "" {
+			return nil, errors.New("unable to determine artifact store")
+		}
+		storePath = filepath.Join(storeOptions.GraphRoot, "artifacts")
+	}
```

Comment:
This code path is not used. We always supply the libpod storage path (which is correct!) so should we drop that code to avoid pointless code?

Reply:
as a library, I felt that the ability to override the store path might be handy.  if nobody thinks so, im happy to change.

Reply:
talked with paul and we agreed we can add this in the future as needed.  removed the code in question and return an error on an empty string.